pxd_fastpath.c

```diff
@@ -1424,6 +1427,12 @@ void pxd_make_request_fastpath(struct request_queue *q, struct bio *bio)
 		return BLK_QC_RETVAL;
 	}
 
+	if (rw != READ && !write_allowed(pxd_dev->mode)) {
+		printk(KERN_ERR"px device %llu is read only, failing IO.\n", pxd_dev->dev_id);
```

Comment:
All these printks in this fn would be problematic if there are a lot of pending IOs in vfs. Printk could be called for every IO, soft locking up the system. Use pxd_printk or printk_ratelimited.